sktime/forecasting/esrnn.py

```diff
@@ -0,0 +1,247 @@
+
+    season1_length : int
+        Period of season 1
+
+    season2_length : int
+        Period of season 2
+
+    window : int
+        Size of Input window, default=5
+
+    """
+
+    _tags = {
+        # packaging info
+        # --------------
+        "authors": ["Ankit-1204"],
+    }
+
+    def __init__(
+        self,
+        input_shape=1,
+        hidden_size=1,
+        num_layer=1,
+        season1_length=3,
+        season2_length=3,
+        seasonality_type="zero",
+        window=5,
+        stride=1,
+        batch_size=32,
+        epoch=10,
+        optimizer="Adam",
+        criterion="pinball",
+        lr_rate=1e-5,
+    ) -> None:
+        super().__init__()
+        self.input_shape = input_shape
+        self.hidden_size = hidden_size
+        self.num_layer = num_layer
+        self.seasonality_type = seasonality_type
+        self.season1_length = season1_length
+        self.season2_length = season2_length
+        self.window = window
+        self.stride = stride
+        self.batch_size = batch_size
+        self.epoch = epoch
+        self.optimizer = optimizer
+        self.criterion = criterion
+        self.lr_rate = lr_rate
+        if _check_soft_dependencies("torch", severity="none"):
+            import torch
+            import torch.nn as nn
+
+            self.loss_list = {
+                "mse": nn.MSELoss,
+                "cross": nn.CrossEntropyLoss,
+                "l1": nn.L1Loss,
+            }
+            self.opti_list = {"adam": torch.optim.Adam, "sgd": torch.optim.SGD}
+
+    def _get_windows(self, y):
+        length = len(y)
+        x_arr = []
+        y_arr = []
+        for i in range(0, length - self.window - self.horizon + 1, self.stride):
+            inp = y[i : i + self.window]
+            out = y[i + self.window : i + self.window + self.horizon]
+
+            x_arr.append(inp)
+            y_arr.append(out)
+
+        if not x_arr:
+            raise ValueError("Input size to small")
+
+        return np.array(x_arr), np.array(y_arr)
+
+    def _instantiate_optimizer(self):
+        import torch
+
+        if self.optimizer:
+            if self.optimizer.lower() in self.opti_list:
+                return self.opti_list[self.optimizer.lower()](
+                    self.network.parameters(),
+                    lr=self.lr_rate,
+                )
+            else:
+                raise TypeError(
+                    f"Please pass one of {self.opti_list.keys()} for `optimizer`."
+                )
+        else:
+            return torch.optim.Adam(self.network.parameters(), lr=self.lr_rate)
+
+    def _instantiate_criterion(self):
+        if self.criterion:
+            if self.criterion in self.loss_list:
+                return self.loss_list[self.criterion]()
+            else:
+                loss = ESRNN().DefaultLoss()
+                return loss
+        else:
+            # default criterion
+            loss = ESRNN().DefaultLoss()
+            return loss
```

Comment:
```suggestion
    def _instantiate_criterion(self):
        if self.criterion:
            return super()._instantiate_criterion()
        return ESRNN.pinball_loss()
```
also see the comments on loss in network class